src/dvr/dvr_rec.c

```diff
@@ -356,6 +357,179 @@ dvr_sub_episode(const char *id, const char *fmt, const void *aux, char *tmp, siz
+          if (localtime_r(&first_aired, &tm)) {
+            sprintf(date_buf, "%04d", tm.tm_year + 1900);
+          }
+        }
+      }
+    }
+  } else {
+    /* Not a movie */
+    if (episode) {
+      /* Get episode information */
+      epg_episode_number_format(episode,
+                                episode_buf, sizeof(episode_buf),
+                                NULL, "S%02d", NULL, "E%02d", NULL);
+
+      const time_t first_aired = episode->first_aired;
+      if (first_aired) {
+        /* Get as yyyy-mm-dd since programme could be one episode a day/week,
+         * unlike films which only needs the year.
+         */
+        struct tm tm;
+        if (localtime_r(&first_aired, &tm)) {
+          strftime(date_buf, sizeof date_buf, "%F", &tm);
+        }
+      }
+    }
+  }
+
+  /* Now we have all our data in place so combine it.
+   * This is based on examples in:
+   * http://kodi.wiki/view/Naming_video_files/Movies
+   * http://kodi.wiki/view/TV_Shows_(Video_Library)
+   */
+
+  size_t offset = 0;
+
+  if (is_movie) {
+    /* TV movies are probably best saved in one folder rather than
+     * multiple folders since video players such as Kodi can download
+     * artwork and information for them anyway and it makes deleting
+     * and moving them easier since they get tracked by inotify on
+     * just the one directory.
+     *
+     * Example format below:
+     *   "tvmovies/title (yyyy)"            (with genre_subdir)
+     *   "title (yyyy)"                     (without genre_subdir)
+     *   "title"                            (without genre_subdir, no airdate)
+     */
+    if (with_genre_subdir)   tvh_strlcatf(tmp, tmplen, offset, "tvmovies/");
+    if (*title_buf)          tvh_strlcatf(tmp, tmplen, offset, "%s", title_buf);
+    /* Movies don't have anything relevant in sub-titles field so
+     * anything there should be ignored. I think some channels store a
+     * translated movie name there (title=original movie name,
+     * subtitle=local language name for movie), but only use title
+     * since scrapers only handle one title.
+     */
+    // if (*subtitle_buf) tvh_strlcatf(tmp, tmplen, offset, " - %s", subtitle_buf);
+    if (*date_buf)  tvh_strlcatf(tmp, tmplen, offset, " (%s)", date_buf);
+  } else {
+    /* TV shows have to go in separate directories based on their title in
+     * order to be scraped properly.
+     * We put the episode number before the subtitle to make it easier
+     * to see if we are missing episodes when you do ls.
+     *
+     * Example formats below:
+     *   "tvshows/title/title - S01E02 - subtitle" (with genre_subdir)
+     *   "title - S01E02 - subtitle"               (without genre_subdir)
+     *   "title - subtitle_2001-05-04"             (without genre_subdir, long running show)
+     *   "title - subtitle"                        (without genre_subdir, no epg info on show)
+     */
+    if (with_genre_subdir) tvh_strlcatf(tmp, tmplen, offset, "tvshows/");
```

Comment:
'tvshows' should be probably configurable through 'DVR Profiles'.

Reply:
Good idea. I've taken a quick look and it should be easy enough to do since we have access to the config via the aux variable. I'll make the change in a couple of days.